fe/fe-core/src/main/java/com/starrocks/analysis/SetPassVar.java

```diff
@@ -84,8 +83,7 @@ public void analyze() {
         }
 
         // 2. No user can set password for root expect for root user itself
-        if (userIdent.getQualifiedUser().equals(Auth.ROOT_USER)
-                && !ClusterNamespace.getNameFromFullName(ctx.getQualifiedUser()).equals(Auth.ROOT_USER)) {
+        if (userIdent.getQualifiedUser().equals(Auth.ROOT_USER)) {
             throw new SemanticException("Can not set password for root user, except root itself");
```

Comment:
check ctx user identification might should be reserved， so that root user can set its password

Reply:
You are right